pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -56,7 +61,8 @@ func discoverKubeAPIServerEndpoint() (string, error) {
 }
 
 // NewNodeConfig creates a new instance of nodeConfig to be used by the caller.
-func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM) (*nodeConfig, error) {
+func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM,
+	clusterServiceCIDR string) (*nodeConfig, error) {
```

Comment:
You need to check the `clusterServiceCIDR` here too. I realize this check is being done in multiple places but given that this is an exported function it could potentially get a bad input. Maybe a common helper for the CIDR check?